networking_cisco/plugins/cisco/cfg_agent/device_drivers/asr1k/aci_asr1k_routing_driver.py

```diff
@@ -395,3 +413,58 @@ def _set_snat_pools_from_hosting_info(self, ri, gw_port, is_delete):
     def _set_nat_pool(self, ri, gw_port, is_delete):
         if gw_port['hosting_info'].get('snat_subnets'):
             self._set_snat_pools_from_hosting_info(ri, gw_port, is_delete)
+
+    def _handle_external_gateway_added_normal_router(self, ri, ext_gw_port):
+        super(AciASR1kRoutingDriver,
+              self)._handle_external_gateway_added_normal_router(ri,
+                  ext_gw_port)
+        # Global conifguration parameters are passed using the global
+        # router. Check to see if there is any global config, and if it's
+        # not already configured, add it in
+        g_configs = ext_gw_port['hosting_info'].get('global_config')
+        if g_configs and isinstance(g_configs, list):
+            self._set_global_config(ri, ext_gw_port, g_configs)
+
+    def _set_global_config(self, ri, port, g_configs):
+        for g_config in g_configs:
+            asr_str = [snippets.GLOBAL_CONFIG_PREFIX]
+            self._config_cmd(ri, port, asr_str, g_config)
+            asr_str.append(snippets.GLOBAL_CONFIG_POSTFIX)
+            self._edit_running_config(''.join(asr_str), 'SET_GLOBAL_CONFIG')
+
+    def _remove_global_config(self, ri, port, g_configs):
+        for g_config in g_configs:
+            asr_str = [snippets.GLOBAL_CONFIG_PREFIX]
+            self._config_cmd(ri, port, asr_str, g_config, remove=True)
+            asr_str.append(snippets.GLOBAL_CONFIG_POSTFIX)
+            self._edit_running_config(''.join(asr_str), 'REMOVE_GLOBAL_CONFIG')
+
+    def _replace_template_vars(self, ri, port, config):
+        kwargs = {}
+        for key in self._template_dict.keys():
+            if '{' + key + '}' in config:
+                kwargs[key] = self._template_dict[key](ri, port, config)
+        return config.format(**kwargs)
+
+    def _config_cmd(self, ri, port, asr_str, g_config, remove=False):
+        if isinstance(g_config, list):
+            for config in g_config:
+                self._config_cmd(ri, port, asr_str, config, remove)
+        else:
+            if remove:
+                snippet = snippets.REMOVE_GLOBAL_CONFIG
+            else:
+                snippet = snippets.SET_GLOBAL_CONFIG
+            asr_str.append(snippet %
+                (self._replace_template_vars(ri, port, g_config)))
+
+    def _set_vrf(self, ri, port, config):
+        return self._get_vrf_name(ri)
+
+    def _set_vrf_pid(self, ri, port, config):
```

Comment:
since this vrf_pid will be used as a ospf router ID. Is this vrf -> router ID mapping 1 to 1? We need to make sure this otherwise there will be conflicts. Pls also put a TBD here saying that we should come up with a better way to produce a unique vrf_pid here as currently it just truncates the vrf ID to 16 bits so potentially there could be clash....